src/azure-cli-core/azure/cli/core/mock.py

```diff
@@ -26,6 +27,10 @@ def __init__(self, commands_loader_cls=None, random_config_dir=False, **kwargs):
 
         if random_config_dir:
             config_dir = os.path.join(GLOBAL_CONFIG_DIR, 'dummy_cli_config_dir', random_string())
+            # Knack prioritizes the AZURE_CONFIG_DIR env over the config_dir param, and other functions may call
+            # get_config_dir directly. We need to set the env to make sure the config_dir is used.
+            self.env_patch = patch.dict('os.environ', {'AZURE_CONFIG_DIR': config_dir})
```

Comment:
`os.environ` should suffice:

```suggestion
            self.env_patch = patch.dict(os.environ, {'AZURE_CONFIG_DIR': config_dir})
```